### ISSUE TITLE
Add `notifyDayRangeChanged` and `notifyVisibleDaysChanged`

### ISSUE THREAD

**Issue by dgudim** [kizitonwose/Calendar#421]
### Feature
Currently the library only provides functions to update current month, one date or entire calendar, but sometimes i need to to update a range of days or all visible days (not only current month). Would it be possible to add such methods? Thanks in advance

### Tech (if applicable)
Add any other information that could help the development of this feature.

**Comment by kizitonwose** [comment]
`notifyCalendarChanged()` calls ` notifyItemRangeChanged(0, itemCount)` internally which only rebinds the currently visible items since the non-visible items will be rebound anyways when they come on screen. I think in cases where you have multiple dates across several months, it is best (and safe!) to use `notifyCalendarChanged()`

**Comment by dgudim** [comment]
Thanks, but what about `notifyDayRangeChanged` some of my events span 3 dates and when I update event color I want to only update those dates

**Comment by kizitonwose** [comment]
Do you have events for in/out dates? There is a method overload that only takes the `LocalDate` and adds `MonthDate` as the position internally.

**Comment by dgudim** [comment]
Yes, some events may extend to in/out date but if I call that overloaded method it will update MonthDate associated with that day when it should update in/out one, so I first need to determine the type of date. Am I getting something wrong?

**Comment by kizitonwose** [comment]
No, you are absolutely correct. Most implementations don't put events in the in/out dates so I thought you had the same. But yep, if you have it the way you explained, then you can't use the overload. 

I see the use for a `notifyDayRangeChanged()` method here but I think it will add more confusion than benefit. For example, the library has to assume that the user passes in the correct day range, or we have to do a bunch of internal validations to ensure the user model accuracy. 

For example, if the user wants to reload date 15 with position inDate and this does not exist, currently, we do nothing, but with `notifyDayRangeChanged`, not sure if we should find the next valid date and reload until the specified end range or just abort the action. 

Do you see any concerns calling `notifyCalendarChanged` currently? If your day bind logic is light, which it should be since it is called multiple times, then `notifyCalendarChanged` would be the way to go.

**Comment by dgudim** [comment]
I totally get your point, not having `notifyDayRangeChanged` is not critical, so I'll use `notifyCalendarChanged`. Thanks for help and clarification :)

**Comment by kizitonwose** [comment]
That looks really nice. Also, yep, keep using `notifyCalendarChanged` ✅

**Comment by kizitonwose** [comment]
I think this can be closed. Feel free to reopen if you need help with anything else.
